bigint/src/lib.rs

```diff
@@ -1852,57 +1873,115 @@ impl serde::Deserialize for BigUint {
-                                                  (893871739, 7), // 19
-                                                  (1280000000, 7), // 20
-                                                  (1801088541, 7), // 21
-                                                  (2494357888, 7), // 22
-                                                  (3404825447, 7), // 23
-                                                  (191102976, 6), // 24
-                                                  (244140625, 6), // 25
-                                                  (308915776, 6), // 26
-                                                  (387420489, 6), // 27
-                                                  (481890304, 6), // 28
-                                                  (594823321, 6), // 29
-                                                  (729000000, 6), // 30
-                                                  (887503681, 6), // 31
-                                                  (1073741824, 6), // 32
-                                                  (1291467969, 6), // 33
-                                                  (1544804416, 6), // 34
-                                                  (1838265625, 6), // 35
-                                                  (2176782336, 6) /* 36 */];
 
-    assert!(2 <= radix && radix <= 36, "The radix must be within 2...36");
-    BASES[radix as usize]
+    match big_digit::BITS {
+        32  => {
+            const BASES: [(u32, usize); 37] = [(0, 0), (0, 0),
+                (0,						0), // 2
+                (3486784401,			20), // 3
+                (0,						0), // 4
+                (1220703125,			13), // 5
+                (2176782336,			12), // 6
+                (1977326743,			11), // 7
+                (0,						0), // 8
+                (3486784401,			10), // 9
+                (1000000000,			9), // 10
+                (2357947691,			9), // 11
+                (429981696,				8), // 12
+                (815730721,				8), // 13
+                (1475789056,			8), // 14
+                (2562890625,			8), // 15
+                (0,						0), // 16
+                (410338673,				7), // 17
+                (612220032,				7), // 18
+                (893871739,				7), // 19
+                (1280000000,			7), // 20
+                (1801088541,			7), // 21
+                (2494357888,			7), // 22
+                (3404825447,			7), // 23
+                (191102976,				6), // 24
+                (244140625,				6), // 25
+                (308915776,				6), // 26
+                (387420489,				6), // 27
+                (481890304,				6), // 28
+                (594823321,				6), // 29
+                (729000000,				6), // 30
+                (887503681,				6), // 31
+                (0,						0), // 32
+                (1291467969,			6), // 33
+                (1544804416,			6), // 34
+                (1838265625,			6), // 35
+                (2176782336,			6) /* 36 */
```

Comment:
Please use spaces inside this table for alignment, rather than tabs, and make that final 36 comment consistent.